explorer/mempool.go

```diff
@@ -208,11 +194,24 @@ func (exp *explorerUI) storeMempoolInfo() (lastBlockHash string, lastBlock int64
 	return
 }
 
-func voteForLastBlock(blockHash, validationHash string) bool {
-	if blockHash != "" && validationHash != blockHash {
-		return false
+func addVoteIndex(tx *MempoolTx, txindexes map[int64]map[string]int) {
+	if idxs, ok := txindexes[tx.VoteInfo.Validation.Height]; ok {
+		if idx, ok := idxs[tx.VoteInfo.TicketSpent]; ok {
+			tx.VoteInfo.MempoolTicketIndex = idx
+		} else {
+			idx := len(idxs) + 1
+			idxs[tx.VoteInfo.TicketSpent] = idx
+			tx.VoteInfo.MempoolTicketIndex = idx
+		}
+	} else {
+		idxs := make(map[string]int)
+		idxs[tx.VoteInfo.TicketSpent] = 1
+		txindexes[tx.VoteInfo.Validation.Height] = idxs
+		tx.VoteInfo.MempoolTicketIndex = 1
 	}
-	return true
+}
+func voteForLastBlock(blockHash, validationHash string) bool {
+	return !(blockHash != "" && validationHash != blockHash)
```

Comment:
This is a little hard to read with the double negatives.  Perhaps an OR instead.

     blockHash == "" || validationHash == blockHash